src/common/hooks/useInjectUserChanges.ts

```diff
@@ -31,7 +31,7 @@ export function useInjectUserChanges(options?: Options) {
   const changes = useUserChanges();
 
   useEffect(() => {
-    if (changes && options?.overwrite === false) {
```

Comment:
@beganovich I don't think this ever worked correctly. David reported that the speed of collapsing/expanding sidebar is really slow - the reason for that is exactly this logic. The changes object is constantly being injected again because this if statement is almost never true.

By default, the changes object used in this if statement is an empty object ({}), it is never null/undefined. So, that part of the condition is always true, while the second part is equal to false only if the overwrite is passed through options. So, instead of comparing types also, we just want to check if the overwrite is set to true.

The overwrite is never passed as a true value, but also I don't think we ever need to overwrite changes over the changes we currently have.